src/server/middleware/blob/put_blob_upload.go

```diff
@@ -15,17 +15,54 @@
 package blob
 
 import (
-	"net/http"
-	"strconv"
-
+	"fmt"
+	"github.com/goharbor/harbor/src/controller/blob"
+	"github.com/goharbor/harbor/src/lib/errors"
 	"github.com/goharbor/harbor/src/lib/log"
+	blob_models "github.com/goharbor/harbor/src/pkg/blob/models"
 	"github.com/goharbor/harbor/src/pkg/distribution"
 	"github.com/goharbor/harbor/src/server/middleware"
+	"github.com/goharbor/harbor/src/server/middleware/requestid"
+	"net/http"
+	"strconv"
 )
 
 // PutBlobUploadMiddleware middleware to create Blob and ProjectBlob after PUT /v2/<name>/blobs/uploads/<session_id> success
 func PutBlobUploadMiddleware() func(http.Handler) http.Handler {
-	return middleware.AfterResponse(func(w http.ResponseWriter, r *http.Request, statusCode int) error {
+
+	before := middleware.BeforeRequest(func(r *http.Request) error {
+		v := r.URL.Query()
+		digest := v.Get("digest")
+
+		if digest == "" {
+			log.Warningf(fmt.Sprintf("the put blob request has no digest in query, %s", r.URL.String()))
+			return errors.New(nil).WithMessage(fmt.Sprintf("the put blob request has no digest in query, %s", r.URL.String()))
```

Comment:
Will this yield a 500?  It should be a 4xx

Reply:
return 400 instead.